src/component/feature/FeatureStrategy/FeatureStrategyEdit/FeatureStrategyEdit.tsx

```diff
@@ -0,0 +1,150 @@
+import { useRequiredQueryParam } from 'hooks/useRequiredQueryParam';
+import { useRequiredPathParam } from 'hooks/useRequiredPathParam';
+import useFeatureStrategyApi from '../../../../hooks/api/actions/useFeatureStrategyApi/useFeatureStrategyApi';
+import { formatUnknownError } from 'utils/format-unknown-error';
+import { useHistory } from 'react-router-dom';
+import useToast from '../../../../hooks/useToast';
+import { IFeatureStrategy, IStrategyPayload } from 'interfaces/strategy';
+import { UPDATE_FEATURE_STRATEGY } from 'component/providers/AccessProvider/permissions';
+import AccessContext from 'contexts/AccessContext';
+
+export const FeatureStrategyEdit = () => {
+    const projectId = useRequiredPathParam('projectId');
+    const featureId = useRequiredPathParam('featureId');
+    const environmentId = useRequiredQueryParam('environmentId');
+    const strategyId = useRequiredQueryParam('strategyId');
+
+    const [strategy, setStrategy] = useState<Partial<IFeatureStrategy>>({});
+    const { updateStrategyOnFeature, loading } = useFeatureStrategyApi();
+    const { feature, refetchFeature } = useFeature(projectId, featureId);
+    const { setToastData, setToastApiError } = useToast();
+    const { hasAccess } = useContext(AccessContext);
+    const { uiConfig } = useUiConfig();
+    const { unleashUrl } = uiConfig;
+    const { push } = useHistory();
+
+    useEffect(() => {
+        const savedStrategy = feature.environments
+            .flatMap(environment => environment.strategies)
+            .find(strategy => strategy.id === strategyId);
+        setStrategy(prev => ({ ...prev, ...savedStrategy }));
+    }, [strategyId, feature]);
+
+    const onSubmit = async () => {
+        try {
+            await updateStrategyOnFeature(
+                projectId,
+                featureId,
+                environmentId,
+                strategyId,
+                createStrategyPayload(strategy)
+            );
+            setToastData({
+                title: 'Strategy updated',
+                type: 'success',
+                confetti: true,
+            });
+            refetchFeature();
+            push(formatFeaturePath(projectId, featureId));
+        } catch (error: unknown) {
+            setToastApiError(formatUnknownError(error));
+        }
+    };
+
+    // Wait until the strategy has loaded before showing the form.
+    if (!strategy.id) {
+        return null;
+    }
+
+    return (
+        <FormTemplate
+            title="Edit feature strategy"
+            description={featureStrategyHelp}
+            documentationLink={featureStrategyDocsLink}
+            formatApiCode={() =>
+                formatUpdateStrategyApiCode(
+                    projectId,
+                    featureId,
+                    environmentId,
+                    strategy,
+                    unleashUrl
+                )
+            }
+        >
+            <FeatureStrategyContainer
```

Comment:
In previous form we've used the convention of ending with Form and used hooks to share functionality. Could we do the same here?

Reply:
Good point, that's a better name. I don't think there's much more to share between Create and Edit (other than what's already shared through `FeatureStrategyForm`).